drivers/spi/spi_gecko_eusart.c

```diff
@@ -0,0 +1,325 @@
+/*
+ * Copyright (c) 2024 Daikin Comfort Technologies North America, Inc.
+ * Copyright (c) 2019 Christian Taedcke <hacking@taedcke.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT silabs_gecko_spi_eusart
+
+#define LOG_LEVEL CONFIG_SPI_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(spi_gecko_eusart);
+#include "spi_context.h"
+
+#include <zephyr/sys/sys_io.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/spi.h>
+#include <soc.h>
+
+#include "em_cmu.h"
+#include "em_eusart.h"
+
+#include <stdbool.h>
+#include <zephyr/drivers/pinctrl.h>
+
+#if DT_NODE_HAS_PROP(n, peripheral_id)
+#define CLOCK_EUSART(id) _CONCAT(cmuClock_EUSART, id)
+#define GET_GECKO_EUSART_CLOCK(n) CLOCK_EUSART(DT_INST_PROP(n, peripheral_id))
+#else
+#if (EUSART_COUNT == 1)
```

Comment:
If `peripheral-id` is made required in the devicetree binding, one can avoid doing this in two different ways.

Reply:
Done, I marked peripheral-id as required and removed that part of the code.

Thanks for your comment!